adam-core/src/main/scala/org/bdgenomics/adam/rdd/sequence/SequenceRDD.scala

```diff
@@ -0,0 +1,330 @@
+    IntervalArray(rdd, SequenceArray.apply(_, _))
+  }
+
+  def union(rdds: SequenceRDD*): SequenceRDD = {
+    val iterableRdds = rdds.toSeq
+    SequenceRDD(rdd.context.union(rdd, iterableRdds.map(_.rdd): _*),
+      iterableRdds.map(_.sequences).fold(sequences)(_ ++ _))
+  }
+
+  /**
+   * Slice the sequences in this RDD to the specified maximum length.
+   *
+   * @param maximumLength Maximum length.
+   * @return Returns a new SliceRDD from the sequences in this RDD sliced
+   *    to the specified maximum length.
+   */
+  def slice(maximumLength: Long): SliceRDD = {
+    def sliceSequence(sequence: Sequence): Seq[Slice] = {
+      val slices: MutableList[Slice] = MutableList()
+
+      val sb = Slice.newBuilder
+        .setName(sequence.getName)
+        .setDescription(sequence.getDescription)
+        .setAlphabet(sequence.getAlphabet)
+        .setSequence(sequence.getSequence)
+        .setStrand(Strand.INDEPENDENT)
+        .setTotalLength(sequence.getLength)
+        .setAttributes(sequence.getAttributes)
+
+      var index = 0
+      var count = (sequence.getLength / maximumLength).toInt
+      if (sequence.getLength % maximumLength != 0) count += 1
+      for (start <- 0L until sequence.getLength by maximumLength) {
+        val end = math.min(sequence.getLength, start + maximumLength)
+        slices += sb
+          .setStart(start)
+          .setEnd(end)
+          .setLength(end - start)
+          .setSequence(sequence.getSequence.substring(start.toInt, end.toInt))
+          .setIndex(index)
+          .setSlices(count)
+          .build()
+        index += 1
+      }
+      slices
+    }
+    SliceRDD(rdd.flatMap(sliceSequence), sequences)
+  }
+
+  /**
+   * Slice the specified sequence overlapping the specified region.
+   *
+   * @param region Region to overlap.
+   * @return Returns a new Slice from the sequence overlapping the specified region.
+   */
+  private def slice(sequence: Sequence, region: ReferenceRegion): Slice = {
+    // region may be open-ended
+    val end = math.min(sequence.getLength, region.end)
+    Slice.newBuilder()
+      .setName(sequence.getName)
+      .setDescription(sequence.getDescription)
+      .setAlphabet(sequence.getAlphabet)
+      .setSequence(sequence.getSequence.substring(region.start.toInt, end.toInt))
+      .setLength(end - region.start)
+      .setTotalLength(sequence.getLength)
+      .setStart(region.start)
+      .setEnd(end)
+      .setStrand(region.strand) // perhaps Sequence should have strand?
+      .setAttributes(sequence.getAttributes)
+      .build()
+  }
+
+  /**
+   * Slice the sequences in this RDD overlapping the specified region.
+   *
+   * @param region Region to overlap.
+   * @return Returns a new SliceRDD from the sequences in this RDD sliced
+   *    to overlap the specified region.
+   */
+  def slice(region: ReferenceRegion): SliceRDD = {
+    SliceRDD(filterByOverlappingRegion(region).rdd.map(sequence => slice(sequence, region)))
```

Comment:
`filterByOverlappingRegion` calls &rarr; `replaceRdd` calls &rarr; `copy`
or perhaps I'm missing what you're saying?